src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1522,6 +1522,22 @@ static bool processSubscriptions
                                                  tenant,
                                                  xauthToken))
     {
+      long long rightNow = getCurrentTime();
+
+      //
+      // If broker running without subscription cache, put lastNotificationTime and count in DB 
+      //
+      if (mongoSubCacheActive == false)
+      {
+        BSONObj query  = BSON("_id" << OID(mapSubId));
+        BSONObj update = BSON("$set" <<
+                              BSON(CSUB_LASTNOTIFICATION << rightNow) <<
+                              "$inc" << BSON(CSUB_COUNT << 1));
+        
+        collectionUpdate(getSubscribeContextCollectionName(tenant), query, update, false, err);
```

Comment:
Old code in processSubscriptions()

```
    {
     ...
      if (collectionUpdate(getSubscribeContextCollectionName(tenant), query, update, false, err))
      {
        // Happy path
        // ...
      else
      {
        ret = false;
      }
    }
```

Thus I think the `ret = false` part in the case of fail should be recovered.